src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/References/ReferenceCleanupService.cs

```diff
@@ -0,0 +1,252 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using EnvDTE;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Build.Exceptions;
+using Microsoft.VisualStudio.ProjectSystem.VS.References.Roslyn;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.References
+{
+    [Export(typeof(IReferenceCleanupService))]
```

Comment:
How does Roslyn get their hands on this service?

Reply:
This service is per "configuration" but the API seems to indicate there's a single version of the interface for the entire solution (because the APIs it provides take projectPaths). Which is it?

Reply:
I think this is supposed to be a normal MEF (non-CPS MEF component) in which case you can't import ConfiguredProject. You should instead be importing IProjectServiceAccessor and getting IProjectService from it.

Reply:
done